odoo/addons/base/res/res_font.py

```diff
@@ -18,6 +20,45 @@
 """
 _logger = logging.getLogger(__name__)
 
+CustomTTFonts = []
+
+# Search path for TTF files, in addition of rl_config.TTFSearchPath
+TTFSearchPath = [
+            '/usr/share/fonts/truetype', # SuSE
+            '/usr/share/fonts/dejavu', '/usr/share/fonts/liberation', # Fedora, RHEL
+            '/usr/share/fonts/truetype/*','/usr/local/share/fonts' # Ubuntu,
+            '/usr/share/fonts/TTF/*', # Mandriva/Mageia
+            '/usr/share/fonts/TTF', # Arch Linux
+            '/usr/lib/openoffice/share/fonts/truetype/',
+            '~/.fonts',
+            '~/.local/share/fonts',
+
+            # mac os X - from
+            # http://developer.apple.com/technotes/tn/tn2024.html
+            '~/Library/Fonts',
+            '/Library/Fonts',
+            '/Network/Library/Fonts',
+            '/System/Library/Fonts',
+
+            # windows
+            'c:/winnt/fonts',
+            'c:/windows/fonts'
+]
+
+def list_all_sysfonts():
+    """
+        This function returns list of font directories of system.
+    """
+    filepath = []
+
+    # Perform the search for font files ourselves, as reportlab's
+    # TTFOpenFile is not very good at it.
+    searchpath = list(set(TTFSearchPath + rl_config.TTFSearchPath))
```

Comment:
assuming the search path is ordered, that's going to be an issue as the `set()` call will deduplicate… and remove the ordering.